File: HandZ6Api/src/main/java/org/example/handz6api/model/EventoTiro.java
// model/EventoTiro.java
package org.example.handz6api.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.example.handz6api.model.enums.ResultadoTiro;
import org.example.handz6api.model.enums.TipoAtaque;
import org.example.handz6api.model.enums.ZonaPorteria;

@Getter
@Setter
@Entity
@Table(name = "evento_tiro")
public class EventoTiro {

    @Id
    @Column(name = "evento_id")
    private Long eventoId;

    @OneToOne
    @MapsId
    @JoinColumn(name = "evento_id")
    private Evento evento;

    @Enumerated(EnumType.STRING)
    @Column(name = "zona", nullable = false)
    private ZonaPorteria zona;

    @Enumerated(EnumType.STRING)
    @Column(name = "resultado", nullable = false)
    private ResultadoTiro resultado;

    @Enumerated(EnumType.STRING)
    @Column(name = "tipo_ataque", nullable = false)
    private TipoAtaque tipoAtaque;

    @ManyToOne
    @JoinColumn(name = "portero_id")
    private Jugador portero;


}

File: HandZ6/app/src/main/java/com/example/handz6/data/Entities/EventoTiro.kt
// entities/EventoTiro.kt
package com.example.handz6.data.local.entities

import androidx.room.*
import com.example.handz6.data.Entities.Evento
import com.example.handz6.data.Entities.Jugador

import com.example.handz6.data.local.enums.ResultadoTiro
import com.example.handz6.data.local.enums.TipoAtaque
import com.example.handz6.data.local.enums.ZonaPorteria

@Entity(
    tableName = "evento_tiro",
    foreignKeys = [
        ForeignKey(
            entity = Evento::class,
            parentColumns = ["id"],
            childColumns = ["evento_id"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Jugador::class,
            parentColumns = ["id"],
            childColumns = ["portero_id"],
            onDelete = ForeignKey.SET_NULL
        )
    ],
    indices = [
        Index(value = ["portero_id"])
    ]
)
data class EventoTiro(
    @PrimaryKey
    @ColumnInfo(name = "evento_id")
    val eventoId: Long,

    @ColumnInfo(name = "zona")
    val zona: ZonaPorteria,

    @ColumnInfo(name = "resultado")
    val resultado: ResultadoTiro,

    @ColumnInfo(name = "tipo_ataque")
    val tipoAtaque: TipoAtaque,

    @ColumnInfo(name = "portero_id")
    val porteroId: Long? = null
)